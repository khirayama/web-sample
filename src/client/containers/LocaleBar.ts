import { Dispatch } from 'redux';
import { connect } from 'react-redux';

import { LocaleBar as Component } from 'client/presentations/components/LocaleBar';
import { changeLocale } from 'client/actions';

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    onEnglishLocaleClick: () => {
      dispatch(changeLocale('en'));
    },
    onJapaneseLocaleClick: () => {
      dispatch(changeLocale('ja'));
    },
  };
};

export const LocaleBar = connect(
  null,
  mapDispatchToProps,
)(Component);
