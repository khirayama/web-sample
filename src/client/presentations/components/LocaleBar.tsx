import * as React from 'react';

type Props = {
  onEnglishLocaleClick: () => void;
  onJapaneseLocaleClick: () => void;
};

export function LocaleBar(props: Props) {
  return (
    <ul>
      <li onClick={props.onEnglishLocaleClick}>English</li>
      <li onClick={props.onJapaneseLocaleClick}>日本語</li>
    </ul>
  );
}
