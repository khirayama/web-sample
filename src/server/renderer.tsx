import express from 'express';
import * as React from 'react';
import * as ReactDOMServer from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import * as styled from 'styled-components';
import ReactHelmet from 'react-helmet';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import { reducer } from 'client/reducers';
import { renderFullPage } from 'server/renderFullPage';
import { Application } from 'client/containers/Application';

export function get(req: express.Request, res: express.Response) {
  const context = {};
  const store = createStore(reducer);
  const sheet = new styled.ServerStyleSheet();
  const body = ReactDOMServer.renderToString(
    sheet.collectStyles(
      <StaticRouter location={req.url} context={context}>
        <Provider store={store}>
          <Application />
        </Provider>
      </StaticRouter>,
    ),
  );
  const preloadedState = store.getState();
  const helmetContent = ReactHelmet.renderStatic();
  const meta = `
      ${helmetContent.meta.toString()}
      ${helmetContent.title.toString()}
      ${helmetContent.link.toString()}
    `.trim();
  const style = sheet.getStyleTags();
  // TODO: script file name;
  const scripts = `<script src="/public/index.bundle.js"></script>`;
  const assets = [''];

  res.send(
    renderFullPage({
      meta,
      assets,
      body,
      style,
      scripts,
      preloadedState: JSON.stringify(preloadedState),
    }),
  );
}
