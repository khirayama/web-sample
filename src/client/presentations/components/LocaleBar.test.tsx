import * as React from 'react';
import renderer from 'react-test-renderer';

import { LocaleBar } from './LocaleBar';

it('Renders correctly', () => {
  const tree = renderer.create(<LocaleBar onEnglishLocaleClick={() => {}} onJapaneseLocaleClick={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
