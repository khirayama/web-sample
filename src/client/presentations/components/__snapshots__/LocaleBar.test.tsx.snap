// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<ul>
  <li
    onClick={[Function]}
  >
    English
  </li>
  <li
    onClick={[Function]}
  >
    日本語
  </li>
</ul>
`;
